runtime/lua/vim/lsp/rpc.lua

```diff
@@ -268,14 +287,14 @@ function M.create_read_loop(handle_body, on_no_chunk, on_error)
   end
 end
 
----@class RpcClient
+---@class vim.lsp.rpc.RpcClient
 ---@field message_index integer
----@field message_callbacks table<integer,function>
----@field notify_reply_callbacks table<integer,function>
----@field transport table
----@field dispatchers table
+---@field message_callbacks table<integer, function> dict of message_id to callback
+---@field notify_reply_callbacks table<integer, function> dict of message_id to callback
+---@field transport vim.lsp.rpc.Transport
+---@field dispatchers vim.lsp.rpc.Dispatchers
 
----@class RpcClient
+---@class vim.lsp.rpc.RpcClient
```

Comment:
Since `rpc` is in the name this can be called `vim.lsp.rpc.Client`. Alternatively you could drop `.rpc`.